### PR TITLE
Bug/repair warnings game write step your mission comp moderation test src index

### PR DESCRIPTION
**Pull request by OniWarrior**
In this branch, I resolved three tickets that dealt with several defined but never used warnings through four files.

In this ticket, I resolved two warnings.

./src/components/pages/Gamemode/YourMissionComp.js
Line 61:11: img elements must have an alt prop, either with meaningful text, or an empty string for decorative images

./src/components/pages/ModerationTest/ModerationTest.js
Line 213:11: Headings must have content and the content must be accessible by a screen reader

The first warning was in the YourMissionComp component. There was a redundancy warning with the alt attribute in the img element. The word image was in the text; this is what caused the warning. I simply changed the text to "A boy" and it resolved the issue.

The second warning was an empty h1 element in the ModerationTest component. I checked the codebase to see if this element was potentially going to be used but found no use for this element. I deleted it and the warning was resolved.
Trello card: https://trello.com/c/IH2fFWuK/615-other-warnings-and-error-b

-------------------------------------------------------------------------------------
In this ticket, I resolved several warnings:
./src/index.js
Line 210:11: No duplicate props allowed
Line 211:11: No duplicate props allowed
Line 212:11: No duplicate props allowed

./src/components/pages/GamePlay/GameWriteStep.js
Line 112:11: Redundant alt attribute. Screen-readers already announce img tags as an image. You don’t need to use the words image, photo, or picture (or any specified custom words) in the alt prop

We had three warnings about duplicates in the index.js file. I checked the codebase and as it turns out, there are no duplicates. The real issue was improper syntax with the ProtectedRoute components on lines 210-211-212. One ProtectedRoute tag was not closed and the other ProtectedRoute on line 212 didn't have a tag at all. I closed the first tag on line 210 and added the ProtectedRoute tag to the ProtectedRoute on line 212. That resolved the issues in this file.

In the GameWriteStep.js file, there was an img element on #112 that had the word "image" in the text of the alt attribute. I changed the text to "A boy" and that resolved the warning.

Trello card: https://trello.com/c/yecJDBnN/617-other-warnings-and-error-d

---------------------------------------------------------------------------------------------
In this ticket, I resolved several defined but never used warnings:

./src/components/pages/Gamemode/YourMissionComp.js
Line 1:27: 'useEffect' is defined but never used
Line 5:20: 'Button' is defined but never used
Line 7:8: 'UploaderComp' is defined but never used
Line 8:10: 'tasks' is defined but never used
Line 9:10: 'Gamemode' is defined but never used
Line 10:10: 'Link' is defined but never used
Line 10:16: 'Route' is defined but never used
Line 11:10: 'render' is defined but never used
Line 14:11: 'push' is assigned a value but never used
Line 14:17: 'location' is assigned a value but never used
Line 15:15: 'setsRwd' is assigned a value but never used

In the YourMissionComp.js file, there were several imports that were imported that were never used.
From line 1-11, I deleted the imports that were triggering the warnings. Note: If we need a particular import in the future, we can import what we need when we are ready to implement. I deleted the push and location consts that were not being used on line 14. 

On line 15 there was a state variable and its update function that was declared. The state variable "rwd" was being used throughout the codebase but the update function, setsRwd, was not being used. The individual who coded this component was trying to directly assign values to the properties of the state variable rather than use the update function to update those values properly. It didn't make sense when I saw this. In any case, I deleted the direct assignments on #20,#22,#24,#34#,36,and #38. I then made a call to the update function, setsRwd, and passed the values from props to the individual properties to update the state variable. I made a call on #25-29 and #39-43. 

I understood what their intention was but the way the individual implemented wasn't correct. With setsRwd implemented, the warning was resolved.

Trello card: https://trello.com/c/0PLOlsOR/603-defined-but-never-used-f

----------------------------------------------------------------------------------
PR submission walkthrough: https://drive.google.com/file/d/1Bu4ilCzA0krvrbMkG0IxlG4RgR_L3Qni/view?usp=sharing

### CODE REVIEW

**Comment by OniWarrior** [comment]
I just want to add one comment. The list of commits is showing that six files were changed. I checked them and it looks like the changes I did for a previous PR. I know for a fact I didn't commit those two files. I committed changes to a total of four files. It's not harming anything but I'm not sure why those two files were included in the push to git hub. I just wanted any reviewers to make note of that for clarity in the review.

**Review by just-kato — APPROVED** [review]

**Review by just-kato — APPROVED** [review]

**Review by just-kato — APPROVED** [review]

**Review by just-kato — APPROVED** [review]

**Review by just-kato — DISMISSED** [review]

**Review by just-kato — DISMISSED** [review]
Great work Stephen! I went in and made some cosmetic fixes to the code. There spaces in between the import sections that needed to be cleaned up.

**Review by just-kato — APPROVED** [review]

**Review by just-kato — APPROVED** [review]

**Review by just-kato — APPROVED** [review]

File: src/components/pages/GamePlay/GameDrawStep.js [hunk 1]
```diff
@@ -97,7 +97,7 @@ export default function GameDrawStep(props) {
     });
 
     setFileList(props.fileSubmissionData.drawings);
-  }, []);
+  }, [props.fileSubmissionData.drawings]);
```
Comment (just-kato) [inline src/components/pages/GamePlay/GameDrawStep.js:96]: Change looks good

File: src/components/pages/GamePlay/GameDrawStep.js [hunk 2]
```diff
@@ -103,7 +103,7 @@ export default function GameDrawStep(props) {
     <div id="draw-step">
       <div id="draw" className="gameplay-content">
         <div className="inner-container">
-          <img src={boyImg} alt="Boy Image" className="boy-img" />
+          <img src={boyImg} alt="A boy" className="boy-img" />
```
Comment (just-kato) [inline src/components/pages/GamePlay/GameDrawStep.js:102]: Good job on keeping the alt tag consistent throughout the code

File: src/components/common/UploadDocs.js [hunk 3]
```diff
@@ -114,7 +114,7 @@ export const UploadDocs = ({
     console.log({ fileList, filePreviews }, 'useEffect');
 
     // This is in case I have a filelist ready OUTSIDE this component
-    if (savedFileList != undefined) {
+    if (savedFileList !== undefined) {
```
Comment (just-kato) [inline src/components/common/UploadDocs.js:116]: Great change. However, I am not seeing this this this file error message in any of the three tickets that you've worked on. I see in the https://trello.com/c/AZgcU47M/592-useeffect-missing-dependencies-warnings-a ticket this error. Was this what you were talking about the 6 extra commits?

File: src/components/common/UploadDocs.js [hunk 4]
```diff
@@ -118,7 +118,7 @@ export const UploadDocs = ({
       setFilePreviews(savedFileList);
       setFileList(savedFileList);
     }
-  });
+  }, [fileList, filePreviews, savedFileList]);
```
Comment (just-kato) [inline src/components/common/UploadDocs.js:120]: Great change. However, I am not seeing this this this file error message in any of the three tickets that you've worked on. I see in the https://trello.com/c/AZgcU47M/592-useeffect-missing-dependencies-warnings-a ticket this error. Was this what you were talking about the 6 extra commits?

File: src/components/pages/ModerationTest/ModerationTest.js [hunk 5]
```diff
@@ -210,7 +210,7 @@ const ModerationTest = props => {
             THIS ROUTE IS DEPRECIATED. PLEASE USE/WORK ON{' '}
             <Link to="/admin">`/admin`</Link> INSTEAD OF `/moderation`
           </h1>
-          <h1></h1>
```
Comment (just-kato) [inline src/components/pages/ModerationTest/ModerationTest.js:213]: Nice work!

File: src/components/pages/GamePlay/GameWriteStep.js [hunk 6]
```diff
@@ -109,7 +109,7 @@ export default function GameWriteStep(props) {
     <div id="write-step">
       <div id="write" className="gameplay-content">
         <div className="inner-container">
-          <img src={boyImg} alt="Boy Image" className="boy-img" />
+          <img src={boyImg} alt="A boy" className="boy-img" />
```
Comment (just-kato) [inline src/components/pages/GamePlay/GameWriteStep.js:112]: Again nice work on keeping the alt tag name "Boy Image"consistent

File: src/components/pages/Gamemode/YourMissionComp.js [hunk 7]
```diff
@@ -58,7 +58,7 @@ const YourMissionComp = ({ ...props }) => {
         className={rwd.write === true ? 'rectangle130-yellow' : 'rectangle130'}
       >
         <Row className="btmRow">
-          <img src={LightingKid} />
+          <img src={LightingKid} alt="A lighting kid" />
```
Comment (just-kato) [inline src/components/pages/Gamemode/YourMissionComp.js:67]: great work

File: src/components/pages/Gamemode/YourMissionComp.js [hunk 8]
```diff
@@ -1,4 +1,4 @@
-import React, { useState, useEffect } from 'react';
+import React, { useState } from 'react';
```
Comment (just-kato) [inline src/components/pages/Gamemode/YourMissionComp.js:1]: Nice change.

File: src/components/pages/Gamemode/YourMissionComp.js [hunk 9]
```diff
@@ -2,7 +2,7 @@ import React, { useState } from 'react';
 import UploadDocs from '../../common/UploadDocs';
 import LightingKid from '../../../assets/images/gamemodeimg/LightingKid.png';
 import { Header } from '../../common';
-import { Row, Col, Button } from 'antd';
+import { Row, Col } from 'antd';
```
Comment (just-kato) [inline src/components/pages/Gamemode/YourMissionComp.js:5]: Indeed correct. Button was never implemented

File: src/components/pages/Gamemode/YourMissionComp.js [hunk 10]
```diff
@@ -8,7 +8,7 @@ import UploaderComp from './UploaderComp';
 import { tasks } from '../../../state/actions';
 import { Gamemode } from './index';
 import { Link, Route, useHistory } from 'react-router-dom';
-import { render } from 'react-dom';
+
```
Comment (just-kato) [inline src/components/pages/Gamemode/YourMissionComp.js]: Awesome work. Just make sure you get rid of that extra line spacing to keep the code looking clean.

File: src/components/pages/Gamemode/YourMissionComp.js [hunk 11]
```diff
@@ -10,7 +10,6 @@ import { Gamemode } from './index';
 
 import Thrashbar from './Thrashbar';
 const YourMissionComp = ({ ...props }) => {
-  const { push, location } = useHistory();
```
Comment (just-kato) [inline src/components/pages/Gamemode/YourMissionComp.js:14]: looks good

File: src/components/pages/Gamemode/YourMissionComp.js [hunk 12]
```diff
@@ -4,7 +4,7 @@ import LightingKid from '../../../assets/images/gamemodeimg/LightingKid.png';
 import { Header } from '../../common';
 import { Row, Col } from 'antd';
 import { connect } from 'react-redux';
-import UploaderComp from './UploaderComp';
+
```
Comment (just-kato) [inline src/components/pages/Gamemode/YourMissionComp.js]: Awesome work. Just make sure you get rid of that extra line spacing to keep the code looking clean.

File: src/components/pages/Gamemode/YourMissionComp.js [hunk 13]
```diff
@@ -5,8 +5,6 @@ import { Header } from '../../common';
 import { Row, Col } from 'antd';
 import { connect } from 'react-redux';
 
-import { Gamemode } from './index';
-
```
Comment (just-kato) [inline src/components/pages/Gamemode/YourMissionComp.js]: great job

File: src/components/pages/Gamemode/YourMissionComp.js [hunk 14]
```diff
@@ -13,10 +13,12 @@ const YourMissionComp = ({ ...props }) => {
     draw: false,
     mode: 'single',
   });
+
   const fxd = () => {
     if (props.child.gamemode.write) {
       props.child.gamemode.write = true;
       rwd.write = true;
+      setsRwd(rwd.write);
```
Comment (just-kato) [inline src/components/pages/Gamemode/YourMissionComp.js]: Nice. State needs to revert back to false

File: src/components/pages/Gamemode/YourMissionComp.js [hunk 15]
```diff
@@ -21,6 +21,7 @@ const YourMissionComp = ({ ...props }) => {
       setsRwd(rwd.write);
       props.child.gamemode.draw = false;
       rwd.draw = false;
+      setsRwd(rwd.draw);
```
Comment (just-kato) [inline src/components/pages/Gamemode/YourMissionComp.js]: Again nice work with the state change

File: src/index.js [hunk 16]
```diff
@@ -200,20 +194,16 @@ function App() {
...
-          exact
-          path="/parent/support"
-          component={() => (
-            <SupportPage LoadingComponent={ParentLoadingComponent} />
-          )}
         />
-
+        exact path="/parent/support" component=
+        {() => <SupportPage LoadingComponent={ParentLoadingComponent} />}
+        />
```
Comment (just-kato) [inline src/index.js]: Nice work catching this.

File: src/index.js [hunk 17]
```diff
@@ -129,34 +129,40 @@ function App() {
             <StoryPrompt LoadingComponent={ChildLoadingComponent} />
           )}
         />
+
```
Comment (just-kato) [inline src/index.js]: Spacing in between routes make the code look so much cleaner. Good job

**Review by just-kato — DISMISSED** [review]

File: src/components/pages/Gamemode/YourMissionComp.js (hunk 14 above)
Comment (rhiannonstanford) [reply src/components/pages/Gamemode/YourMissionComp.js]: Great catch here!

File: src/components/common/UploadDocs.js (hunk 3 above)
Comment (OniWarrior) [reply src/components/common/UploadDocs.js:116]: Yes it was. I don't know why these extra commits showed up as part of the push. These were never part of the list of changes I committed. These were the changes I made from my last PR.

**Comment by OniWarrior** [comment]
Yes those that were marked outdated showed up on the list commits during this push. I don't know why because I never staged them during this push.

File: src/components/pages/Gamemode/YourMissionComp.js [hunk 18]
```diff
@@ -1,40 +1,46 @@
...
 const YourMissionComp = ({ ...props }) => {
-  const { push, location } = useHistory();
   const [rwd, setsRwd] = useState({
     read: false,
     write: false,
     draw: false,
     mode: 'single',
   });
+
   const fxd = () => {
```
Comment (rhiannonstanford) [inline src/components/pages/Gamemode/YourMissionComp.js:17]: For the developer that came before you, the name of this function could have benefited from a more clear name.  A name that would help future developers understand the utility of this function at a glance.  Nice work sorting through this code and making sense of it!

File: src/components/pages/Gamemode/YourMissionComp.js [hunk 19]
```diff
@@ -1,40 +1,46 @@
...
+
       props.child.gamemode.read = true;
-      rwd.read = true;
+
+      setsRwd({
+        write: props.child.gamemode.write,
+        draw: props.child.gamemode.draw,
+        read: props.child.gamemode.read,
+      });
     }
```
Comment (rhiannonstanford) [inline src/components/pages/Gamemode/YourMissionComp.js:30]: Excellent job understanding what's happening in the logic here.  You set rwd state properly.  The functionality of this component and page works as expected after your change.  It conditionally renders text and styling based on child selection which is set to rwd state.

File: src/components/pages/GamePlay/GameDrawStep.js [hunk 20]
```diff
@@ -97,13 +93,13 @@ export default function GameDrawStep(props) {
...
     setFileList(props.fileSubmissionData.drawings);
-  }, []);
+  }, [props.fileSubmissionData.drawings]);
 
   return (
     <div id="draw-step">
       <div id="draw" className="gameplay-content">
         <div className="inner-container">
-          <img src={boyImg} alt="Boy Image" className="boy-img" />
+          <img src={boyImg} alt="A boy" className="boy-img" />
```
Comment (rhiannonstanford) [inline src/components/pages/GamePlay/GameDrawStep.js:103]: Great addition of appropriately named alt tag.  Enhances accessibility.

File: src/components/pages/GamePlay/GameWriteStep.js [hunk 21]
```diff
@@ -109,7 +109,7 @@ export default function GameWriteStep(props) {
...
```
Comment (rhiannonstanford) [inline src/components/pages/GamePlay/GameWriteStep.js:113]: Nice correction here.

File: src/index.js [hunk 22]
```diff
@@ -207,6 +207,8 @@ function App() {
           component={() => (
             <ParentDashFaq LoadingComponent={ParentLoadingComponent} />
           )}
+        />
+        <ProtectedRoute
```
Comment (rhiannonstanford) [inline src/index.js:211]: This was another good catch and excellent attention to detail.

**Review by rhiannonstanford — APPROVED** [review]
You implemented solid fixes to the code and nice work understanding the code base and logic as related to YourMissionComp.js.  I did some testing of the page and it's functionality. It is working well as expected with conditional rendering and proper state in place.  Really great work!  My only suggestion for improvement would have been to divide this PR into 3 PRs.  This enhances readability for the reviewer and better segments the bug issues.  Smaller PRs enable the reviewer to go through the code more efficiently.  Overall, excellent work!

**Comment by OniWarrior** [comment]
Apologies to the reviewers. Qirhi you're correct about this PR. I've been a little bit obsessive about clearing out the warnings that I think I did too much for this PR. I promise that the next one isn't going to be as heavy.

**Comment by rhiannonstanford** [comment]
You did excellent work here!  Thank you!

**Comment by OniWarrior** [comment]
This is too big. I'm going to break this up into 3 smaller PRs. Once I'm done I'll close this PR.

**Review by rhiannonstanford — APPROVED** [review]
All changes have been reviewed in previous PR submission.  Page functions as expected.  Excellent work!

**Review by ashtilawat — COMMENTED** [review]
I don't mind merging this PR if it gets one more review.

**Comment by OniWarrior** [comment]
This PR is being closed. The reason being is because it was too big. I broke this apart by submitting 3 smaller PRs that are ready for review.

**Pull request closed without merging**